core/recourse_iterator.py

```diff
@@ -1,9 +1,8 @@
 from typing import Protocol, Sequence
```

Comment:
It would be useful to add, at a minimum, a single-line file header describing what the file does/contains/etc. This goes for the other files, too.

Reply:
We can revisit this later in issue #25